src/com/rabbitmq/client/impl/recovery/AutorecoveringConnection.java

```diff
@@ -428,18 +470,43 @@ private void recoverBlockedListeners() {
         }
     }
 
-    private void recoverConnection() throws IOException, InterruptedException {
-        boolean recovering = true;
-        while (recovering) {
+	//Returns true if the connection was recovered, 
+	//false if application initiated shutdown while attempting recovery.  
+    private boolean recoverConnection() throws IOException, InterruptedException {
```

Comment:
I disagree with the decision to turn a side-effecting method into a boolean. It should be a separate method (or even a condition on `manuallyClosed`).